hawkular-inventory-bus-api/src/main/java/org/hawkular/inventory/bus/api/InventoryAbstractMessage.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.inventory.bus.api;
+
+import org.hawkular.bus.common.AbstractMessage;
+import org.hawkular.inventory.api.model.CanonicalPath;
+import org.hawkular.inventory.json.InventoryJacksonConfig;
+import org.hawkular.inventory.json.mixins.model.CanonicalPathMixin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Pavol Loffay
+ * @since 0.10.0
+ */
+public abstract class InventoryAbstractMessage extends AbstractMessage {
+
+    @Override
+    protected ObjectMapper buildObjectMapperForSerialization() {
+        final ObjectMapper mapper = new ObjectMapper();
+        InventoryJacksonConfig.configure(mapper);
+        mapper.addMixIn(CanonicalPath.class, CanonicalPathMixin.class);
+        return mapper;
+    }
+
+    public static ObjectMapper buildObjectMapperForDeserialization() {
+        /**
+         * @see url(https://stackoverflow.com/questions/19694928/
+         * jackson-jersey-deserialize-exception-for-id-type-id-class-no-such-class)
+         */
+        Thread.currentThread().setContextClassLoader(InventoryAbstractMessage.class.getClassLoader());
```

Comment:
Ok, but in the EE container environment, resetting the context classloader is considered dangerous.

So what if, instead you returned a custom `ObjectMapper` decorator that would set and reset the context classloader in every method? There are only like 300 of them :wink:.

Or maybe even better - this should be fixed in bus that should set and reset the context classloader before and after using this object mapper, as per https://stackoverflow.com/questions/19694928/+jackson-jersey-deserialize-exception-for-id-type-id-class-no-such-class.